Code/skyrim_ui/src/app/services/chat/commands.ts

```diff
@@ -0,0 +1,63 @@
+import { ChatService, MessageTypes } from "../chat.service";
+
+export interface Command {
+  readonly name: string,
+  readonly executor: (args: string[]) => Promise<void>
+}
+
+export class CommandHandler {
+
+  private Help: Command = { name: 'help', executor: async () => {
+    const cmds = [...this.commands.keys()].join(', ');
+    this.chatService.pushSystemMessage('SERVICE.COMMANDS.AVAILABLE_COMMANDS', {cmds});
+  }}
+
+  private PartyChat: Command = { name: 'party', executor: async (args) => {
+    const content = args.join(' ');
+    this.chatService.sendMessage(MessageTypes.PARTY_CHAT, content);
+  }}
+
+  private LocalChat: Command = { name: 'local', executor: async (args) => {
+    const content = args.join(' ');
+    this.chatService.sendMessage(MessageTypes.LOCAL_CHAT, content);
+  }}
+
+  private readonly commands: Map<string, Command>;
+
+  private readonly chatService: ChatService
+
+  public constructor (
+    chatService: ChatService
+  ) {
+    this.chatService = chatService;
+
+    this.commands = new Map<string, Command>();
+    this.register(this.Help);
+    this.register(this.PartyChat);
+    this.register(this.LocalChat);
+  }
+
+  public readonly COMMAND_PREFIX = '/';
+
+  public register(cmd: Command ) {
+    if (!this.commands.has(cmd.name)) {
+      this.commands.set(cmd.name, cmd);
+    }
+  }
+
+  public async tryExecute(input: string) {
+    if (!input.startsWith(this.COMMAND_PREFIX) || input.length < this.COMMAND_PREFIX.length) {
```

Comment:
i think the first check should suffice. Maybe make them case insensitive too

Reply:
agreed. fixed